[Week1] 필수 & 심화 과제

sonms opened the pull request:
## 📌 PR 요약

🫛 작업한 내용
- [x] 필수 과제
- [x] 심화 과제 

 🫛 PR 포인트
- 필수 과제 구현 => SignUp, SignIn, Main

<table>
  <tr>
    <td align="center" width="50%">
      <h3>🎯 필수용</h3>
      <video src="https://github.com/user-attachments/assets/49b53a62-bd70-484d-ab12-a997c2b2f6cb" controls width="100%"></video>
    </td>
    <td align="center" width="50%">
      <h3>🚀 심화용</h3>
      <video src="https://github.com/user-attachments/assets/af64aadf-6965-4175-9072-dff15bb954dc" controls width="100%"></video>
    </td>
  </tr>
</table>


# 📌관련 이슈
- closed #2

Commit:
feat: Update target and compile SDK to 36

Commit:
feat: Add signin and signup activities

- Add `SignInActivity` and `SignUpActivity`.
- Set `SignInActivity` as the launcher activity.
- Add string resources for new activity titles.
- Add drawable assets for password visibility toggle (eye on/off).

Commit:
feat: Add DiveSoptTextField and FormField composables

- Create `DiveSoptTextField`, a custom text field component.
- Create `FormField`, a component that combines a label and `DiveSoptTextField`.
- Implement password visibility toggling in the `FormField` for password inputs.

Commit:
feat: Add signin and signup activities

- Add `SignInActivity` and `SignUpActivity`.
- Move `MainActivity` to the `ui.main` package.
- Implement basic UI and navigation for sign-in and sign-up screens.

Commit:
feat: Add sign-up form validation logic

- Create a `validateSignUpForm` utility function.
- Implement checks for ID, password, and nickname fields to ensure they are not blank.
- Add length validation for ID (6-10 characters) and password (8-12 characters).
- Add a check to ensure the nickname is longer than one character.
- Display a `Toast` message for each specific validation failure.

Commit:
mod: Modify label style in FormField

- Change `fontSize` from 36.sp to 32.sp.
- Remove `fontWeight = FontWeight.Thin`.

Commit:
feat: Add DiveSoptButton composable

- Create `DiveSoptButton`, a custom button component.
- The button fills the maximum width and has a blue background.
- It takes a text string and an onClick lambda as parameters.

Commit:
feat: Add sign in and sign up logic

- Implement sign-in logic to pass user data to `MainActivity` upon successful login.
- Implement sign-up logic to validate form inputs and return user data to `SignInActivity`.
- Update `MainActivity` to receive and display user information from the sign-in process.
- Replace `Button` with the custom `DiveSoptButton` composable in `SignInActivity` and `SignUpActivity`.

Commit:
feat: Add user profile UI to MainActivity

- Implement a user profile section in `MainScreen`.
- Display a profile image, name, and a short bio.
- Add `UserInfoHolder`, a new composable to display labeled user information like ID, Password, Nickname, and Alcohol tolerance.
- Add a new profile image drawable resource.

Commit:
mod: Increase profile image size in MainActivity

- Change the size of the profile image from 36.dp to 48.dp.

sonms committed:
refactor: Relocate components and theme to designsystem module

- Move composables (`DiveSoptButton`, `DiveSoptTextField`, `FormField`) from `ui.core.component` to `core.designsystem.component`.
- Move theme-related files (`Color.kt`, `Type.kt`, `Theme.kt`) from `ui.theme` to `core.designsystem.theme`.
- Update import paths in `MainActivity`, `SignInActivity`, and `SignUpActivity` to reflect the new locations.

sonms committed:
refactor: Remove SignInActivity label from AndroidManifest

- Delete the `android:label` attribute for `SignInActivity`.

Commit:
docs: Add issue and pull request templates

- Add `assignment_template.md` for seminar assignments.
- Add `pull_request_template.md` to standardize pull requests.

Commit:
fix: Update Compose BOM to 2025.10.00

- Upgrade the version of `composeBom` from `2024.09.00` to `2025.10.00` in the version catalog.

Commit:
refactor: Update AndroidManifest settings

- Add `android:windowSoftInputMode="adjustResize"` to `SignUpActivity` to ensure the layout resizes when the keyboard appears.
- Remove the `android:label` attribute from `MainActivity`.

Commit:
feat: Add Modifier extensions for UI interactions

- Create `noRippleClickable` extension to remove the ripple effect from clickable elements.
- Introduce `advancedImePadding` extension to provide more refined keyboard padding behavior.
- Add `NoRippleInteractionSource` to support the implementation of the ripple-free clickable modifier.

Commit:
feat: Add input and output transformations for text fields

- Create `PasswordOutputTransformation` to replace all characters in a password field with asterisks (*).
- Implement `IdInputTransformation` to limit user ID input to a maximum of 10 characters.
- Implement `PasswordInputTransformation` to limit password input to a maximum of 12 characters.
- Implement `NicknameInputTransformation` to limit nickname input to a maximum of 20 characters.

Commit:
refactor: Adopt TextFieldState and enhance FormField composable

- Modify `DiveSoptTextField` and `FormField` to use `TextFieldState` for state management, replacing the previous `value` and `onTextChange` parameters.
- Implement `InputTransformation` in `FormField` to enforce specific rules for ID, Nickname, and Password fields.
- Add `PasswordOutputTransformation` to obscure password text.
- Introduce `imeAction` and `onImeActionPerformed` to `FormField` for better keyboard action handling.
- Remove ripple effect from the password visibility toggle `IconButton`.
- Apply a multi-color gradient brush to the text style in `DiveSoptTextField`.

Commit:
refactor: Remove ripple effect from DiveSoptButton

- Remove the ripple effect on click by setting the `interactionSource` to `NoRippleInteractionSource`.
- Add a `modifier` parameter to allow for external modifications.

Commit:
feat: Add AuthManager for SharedPreferences handling

- Create `AuthManager`, a singleton object to manage user data using `SharedPreferences`.
- Implement `saveUserCredentials` to store user ID, password, nickname, and alcohol level.
- Add getter functions (`getSavedId`, `getSavedPassword`, etc.) to retrieve stored user data.
- Include `clearUserCredentials` to remove all saved data from preferences.
- Add an `init` function to initialize the manager with the application context.

Commit:
feat: Implement auto-login and logout functionality

- Implement an auto-login feature in `SignInActivity` using `AuthManager` to check for saved user credentials.
- Add a logout button to `MainActivity` that clears user credentials and finishes the activity.
- Save user credentials to `AuthManager` upon successful sign-up in `SignUpActivity`.

refactor: Improve UX and state management in auth screens

- Replace `mutableStateOf` with `rememberTextFieldState` for input fields in `SignInActivity` and `SignUpActivity` for better state handling.
- Implement keyboard focus management using `FocusRequester` to move between input fields when the "Next" action is triggered.
- Add `advancedImePadding` and `verticalScroll` to `SignUpScreen` to prevent the keyboard from hiding input fields.
- Make the password visibility state saveable across configuration changes using `rememberSaveable`.
- Replace nullable intent extras with non-nullable strings with default empty values in `MainActivity`.

sohee6989 commented on app/src/main/java/com/sopt/dive/ui/signin/SignInActivity.kt:
```diff
@@ -0,0 +1,242 @@
+            modifier = Modifier
+                .fillMaxWidth()
+                .noRippleClickable(onClick = onSignUpClick),
+            textAlign = TextAlign.Center,
```
현재 '회원가입하기' 텍스트만 클릭이 가능한게 아닌 전체 넓이에 대해 클릭이 가능한데 이 부분은 의도하신 걸까요??

sonms replied:
네 ㅋㅋㅋ 누르기 편하려고 전체 크기를 줬습니다!

sohee6989 commented on app/src/main/java/com/sopt/dive/core/designsystem/component/FormField.kt:
```diff
@@ -0,0 +1,116 @@
+    onVisibilityChange: () -> Unit = {},
+    onImeActionPerformed: () -> Unit = {}
+) {
+    val trailingIcon = if (label == "PASSWORD") {
```
아예 아이콘 자체를 변수로 따로 빼신 이유가 있으실까요??

sonms replied:
https://github.com/37-dive-sopt-android/nahyeon-kim/pull/3#discussion_r2443346234

팟짱님과 같은 생각이에요
만약 뒤에 붙는 아이콘이 현재는 label로 분류 되지만 더 많아질 경우에 
ui에서 사용되는 구조라면 추후 관리가 여려워 질 수 있을 것 같아서
그렇다면 이를 한 곳에서 감지해서 view로 내려주는 건 어떨까? 싶어서 택했어요

sohee6989 commented on app/src/main/java/com/sopt/dive/core/extension/ModifierExtensions.kt:
```diff
@@ -0,0 +1,41 @@
+}
+
+@SuppressLint("AutoboxingStateCreation")
+fun Modifier.advancedImePadding() = composed {
```
덕분에 겪고 있던 문제를 해결했습니다...ㅜㅜㅜ

angryPodo replied:
`@SuppressLint("AutoboxingStateCreation")` 이거 제거해주삼~

sohee6989 commented on app/src/main/java/com/sopt/dive/core/util/NoRippleInteractionSource.kt:
```diff
@@ -0,0 +1,14 @@
+import kotlinx.coroutines.flow.emptyFlow
+
+// 버튼과 같은 Modifier.clickable을 사용하지 않을 때 물결 효과를 지우기용
+class NoRippleInteractionSource : MutableInteractionSource {
```
최근에 버튼 컴포넌트를 Button을 이용하는게 아닌 Text를 이용해서 계속해서 만들어서 이 부분은 고려하지도 못했었는데 Button 이용하는 경우에는 이 부분 까먹지 않아야겠네요~~!!!ㅎㅎ

Commit:
refactor: Generalize FormField and improve keyboard handling

Commit:
refactor: Optimize noRippleClickable modifier extension

- Refactor the `noRippleClickable` modifier to enhance its structure and readability.
- Reorder the parameters in the `clickable` function for better clarity.
- Add commented-out code demonstrating an alternative, performance-optimized implementation using `ModifierNodeElement` for consideration.

Commit:
refactor: Replace `rememberCoroutineScope` with `lifecycleScope`

- Replace `rememberCoroutineScope` with `lifecycleScope` for coroutine management in `MainActivity` and `SignUpActivity`.
- This change aligns coroutine lifetimes with the lifecycle of the respective activities, preventing potential memory leaks.

Commit:
refactor: Remove unused `rememberCoroutineScope` import

- Remove the unnecessary import of `rememberCoroutineScope` from `MainActivity` and `SignUpActivity` as it is no longer in use.

chattymin commented on app/src/main/java/com/sopt/dive/core/designsystem/component/DiveSoptTextField.kt:
```diff
@@ -0,0 +1,94 @@
+fun DiveSoptTextField(
+    state: TextFieldState,
+    placeholder : String,
+    inputTransformation: InputTransformation?,
+    outputTransformation: OutputTransformation?,
```
이 두 값이 필수인 이유가 있나요? 해당 컴포넌트를 활용하는 곳을 보니 null을 기본값으로 넣어주고 있는 것 같습니다.

sonms replied:
앗 그렇네용 기본값을 가지고 있는게 좋겠네요!

chattymin commented on app/src/main/java/com/sopt/dive/core/designsystem/component/FormField.kt:
```diff
@@ -0,0 +1,66 @@
+import com.sopt.dive.core.designsystem.theme.DiveTheme
+
+@Composable
+fun FormField(
+    state: TextFieldState,
+    label: String,
+    placeholder: String,
+    imeAction: ImeAction,
+    modifier: Modifier = Modifier,
+    inputTransformation: InputTransformation? = null,
+    outputTransformation: OutputTransformation? = null,
+    onImeAction: () -> Unit = {},
+    trailingIcon: @Composable (() -> Unit)? = null
+) {
+    Text(
+        text = label,
+        fontSize = 32.sp,
+        modifier = Modifier
+            .fillMaxWidth(),
+        textAlign = TextAlign.Start
+    )
+
+    DiveSoptTextField(
+        modifier = modifier
+            .fillMaxWidth(),
+        state = state,
+        placeholder = placeholder,
+        inputTransformation = inputTransformation,
+        outputTransformation = outputTransformation,
+        trailingIcon = trailingIcon,
+        imeAction = imeAction,
+        onImeAction = onImeAction
+    )
+
+    Spacer(modifier = Modifier.padding(8.dp))
+}
```
해당 FormField를 어느 레이아웃에 넣느냐에 따라 가로정렬, 세로정렬, 겹침 등 다양하게 UI가 달라질 것 같습니다. 내부에 Column으로 묶어두면 어느 레이아웃에 넣더라도 같은 컴포넌트의 UI가 나올 것 같습니다.

그리고 하단의 8dp의 경우 컴포넌트 내부에 넣어야 할까요?

sonms replied:
아 그럼 궁금한게 있습니다! 만약 전체 Column안에만 세워둘 컴포넌트라면 그럼에도 Column으로 감싸야할까요? 그렇게 되면 Column 안에 Column이 되서 미세한 성능에 영향을 끼치지 않을까요?
하단 같은 경우 세로로 쌓아 놓은 컴포넌트 역할이라 여백까지 책임을 가져도 된다고 생각했어요 근데 다른 레이아웃에서 재사용될 수 있다면 빼는게 좋을 것 같네요!

chattymin replied:
말씀해주신 것 처럼 아주 미세한 성능 하락이 되긴 합니다.
근데 정말 미세해서 그정도는 전혀 문제없구요, 그정도의 성능보다는 저는 가독성과 안전성이 더 중요하다고 생각해요.
해당 컴포넌트가 무조건 세로에만 들어간다는 보장을 런타임/컴파일 단계에서 보장할 수 없다면 Column에 감싸는게 맞다고 생각해요.

chattymin commented on app/src/main/java/com/sopt/dive/ui/main/MainActivity.kt:
```diff
@@ -0,0 +1,188 @@
+        enableEdgeToEdge()
+        AuthManager.init(this@MainActivity)
+
+        val userId = intent.getStringExtra("USER_ID") ?: ""
```
`.orEmpty`도 있답니다 ㅎ.ㅎ

sonms replied:
앗 역시 팟짱님 ㅎㅎ 감사합니다 !

chattymin commented on app/src/main/java/com/sopt/dive/data/remote/AuthManager.kt:
```diff
@@ -0,0 +1,64 @@
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+
+object AuthManager {
+    private const val KEY_USER_ID = "user_id"
+    private const val KEY_USER_PASSWORD = "user_password"
+    private const val KEY_USER_NICKNAME = "user_nickname"
+    private const val KEY_USER_ALCOHOL = "user_alcohol"
+
+    private lateinit var appContext: Context
+
+    fun init(context: Context) {
+        appContext = context.applicationContext
+    }
+    private val prefs: SharedPreferences by lazy {
+        appContext.getSharedPreferences("auth_prefs", Context.MODE_PRIVATE)
+    }
```
init 하기 전 하위 함수들을 호출한다면 어떻게 될까요?
Exception이 발생해 앱이 꺼지지 않도록 하기 위한 장치가 있으면 좋을 것 같습니다.
특히 오브젝트니까요.

sonms replied:
맞네요! isInitialized을 추가해서 init을 호출하도록 해야할 것 같네요!

chattymin commented on app/src/main/java/com/sopt/dive/ui/signin/SignInActivity.kt:
```diff
@@ -0,0 +1,269 @@
+    }
+
+    private fun tryAutoLogin() {
+        val saveUserId = AuthManager.getSavedId().takeIf { it.isNotBlank() }
```
early return을 적용하는건 어떨까요?
saveUserId가 값이 비어있다면 하위까지 모두 검사할 필요 없어보여요

sonms replied:
오 그렇네요 좋은 방식 같습니다 바로 반영하겠습니다 ㅎㅎㅋㅋ

chattymin commented on app/src/main/java/com/sopt/dive/ui/signin/SignInActivity.kt:
```diff
@@ -0,0 +1,269 @@
+    onSignInClick: (String, String) -> Unit,
+    onSignUpClick: () -> Unit
+) {
+    val focusManager = LocalFocusManager.current
+
+    var isPasswordVisible by rememberSaveable { mutableStateOf(false) }
+
+    val idText = rememberTextFieldState("")
+
+    val passwordText = rememberTextFieldState("")
```
이 값들을 Screen이 아닌 Route 즉, 상위로 호이스팅 했을때의 이점이 뭔가요?
특히 focusManager의 경우 Route에서 사용하고 있지 않은 것 같습니다.

sonms replied:
아 그렇네요 상태 관리를 위해 사용하려면 focus도 내려줄게 아니라 호이스팅해서 올렸어야 했네요!

chattymin commented on app/src/main/java/com/sopt/dive/ui/signup/util/validateSignUpForm.kt:
```diff
@@ -0,0 +1,40 @@
+import android.content.Context
+import android.widget.Toast
+
+fun validateSignUpForm(
```
if 내부에 하나씩 다 Toast 쓰는게 비효율적인 것 같네요
최적화 가능할 것 같습니다. 아래는 대충 생각한 예시니까 참고만 해주세요

```
val errorMessage = when {
    idText.isBlank() -> "아이디 어쩌구"
    passwordText.isBlank() -> "비번 어쩌구"
    else -> null
}

errorMessage?.let {
    Toast ~
}

return errorMessage != null

```

sonms replied:
넵 참으면서 하겠습니다!

chattymin reviewed (commented):
아이 잘하네요 역시~
중간중간 어려운 / 새로운 개념들은 와비친구들도 같이 이해할수 있게 주석 달아주면 좋을 것 같아요!

sonms committed:
refactor: 코리반영

- Refactor `validateSignUpForm` to use a `when` expression for more concise validation and to display a single Toast message.
- Modify the nickname validation to check for a length greater than one character (`nicknameText.length <= 1`).
- Abstract `moveFocus` and `clearFocus` as lambdas in `SignInScreen` and `SignUpScreen` to decouple them from the `LocalFocusManager`.
- Add a check in `AuthManager` to ensure it is initialized before accessing `SharedPreferences`.
- Add a condition to `tryAutoLogin` in `SignInActivity` to exit early if no ID is saved.
- Simplify nullable string handling in `SignInActivity` using `.orEmpty()`.

sonms committed:
refactor: Remove unused focusManager from SignInScreen

- Delete the `focusManager` parameter from the `SignInScreen` composable call in `SignInActivity`.

sonms committed:
refactor: Remove unused FocusManager import

- Delete the import for `androidx.compose.ui.focus.FocusManager` in `SignInActivity` as it is no longer in use.

hyeminililo commented on app/src/main/java/com/sopt/dive/core/designsystem/component/DiveSoptTextField.kt:
```diff
@@ -0,0 +1,94 @@
+private fun DiveSoptTextFieldPreview() {
+    DiveTheme {
+        val state = TextFieldState()
+        DiveSoptTextField(
```
TextField를 직접 커스텀해서 과제에 맞게 만드신게 굉장히 인상적이네요 🙂

hyeminililo commented on app/src/main/java/com/sopt/dive/core/util/OutputTransformation.kt:
```diff
@@ -0,0 +1,12 @@
+import androidx.compose.foundation.text.input.TextFieldBuffer
+
+object PasswordOutputTransformation : OutputTransformation {
+    override fun TextFieldBuffer.transformOutput() {
```
TextField 내에  visualTransformation 이 속성을 사용하지 않고, 직접 구현하신것 같은데 그 이유가 궁금합니다 !

sonms replied:
이건 제가 아티클을 적어드릴건데 이 코드는 안드로이드에서 새로운 TextField를 만들었어요! text와 onValueChange를 사용하지 않고 TextFieldState를 쓰는 새로운 친구입니다! 
이 친구는 visualTransformation을 파라미터로 가지고 있지않고 InputTransformation과 OutputTransformation을 사용해요! 이 부분에 대해서는

[https://developer.android.com/develop/ui/compose/text/user-input?textfield=state-based&hl=ko#manage-tf-state](https://developer.android.com/develop/ui/compose/text/user-input?textfield=state-based&hl=ko#manage-tf-state)을 읽어보면 좋을 것 같아요!

hyeminililo reviewed (commented):
코드 리뷰를 하면서, 최적화에 대해 많이 고민하신게 느껴졌습니다.  민성님 코드를 통해 단순히 돌아가는 것을 넘어 한 단계 깊게 생각하고 고민하는 시간을 더 가져야겠다고 배웠습니다.  코드 잘 읽었습니다 :)
과제 하시느라 수고하셨습니다 ☺️

oilbeaneda commented:
가시적으로 코드가 어떤기능을 담당하고 있는지도 눈에 잘 들어오고 길지않은코드에 모든 기능이 다 담겨져있음에 박수를 치고갑니다 . 수고많으셨습니다!!

oilbeaneda commented on app/src/main/java/com/sopt/dive/ui/signup/util/validateSignUpForm.kt:
이렇게 회원가입/메인/로그인 뷰에서 모든 기능을 구현하지 않고 각 형태를 제작해서 구현하신게 매우인상깊습니다! 하나의 기능을 구현할때 여러 형태를 나누는 민성님만의 기준이 있으신지 궁금합니답!! 
저는 메인/로그인/회원가입)에서 모든걸 다 담으려했는데 이렇게 나누었을때의 이점이 있을까요?!!

sonms replied:
정말 좋은 질문이에요~! 조금은 어려우신 개념이실 수도 있지만 NowInAndroid 의 구조에 대해 살펴보시면 더욱 좋아요!
현재 제 코드는 Route와 Screen 개념으로 나눠져 있는데 추후 SAA (Single Architecture Activity) 에서 사용될 구조로 Route는 상태 관리, 사이드 이펙트 관리 등을 다룰 곳이라면 Screen은 Route에서 받은 데이터를 사용해서 실제 UI를 구성하는 함수에요 이런 방법을 통해 화면 간 의존성을 관리하고 유지보수성과 테스트 용이성을 높일 수 있어요!! 더 궁금하시다면 [https://github.com/android/nowinandroid](https://github.com/android/nowinandroid) 을 직접 확인하셔도 좋고 댓글 다셔도 좋습니다!

그래서 저는 저 구조의 장점을 위해서 사용했어요!

oilbeaneda commented on app/src/main/java/com/sopt/dive/ui/signin/SignInActivity.kt:
```diff
@@ -0,0 +1,277 @@
+                    SignInRoute(
+                        paddingValues = innerPadding,
+                        onSignInClick = { loginId, loginPassword ->
+                            if ((signUpId == loginId && signUpId.isNotBlank()) && (signUpPassword == loginPassword && signUpPassword.isNotBlank())) {
```
회원가입할때 공백이면 다시 입력하라고해서 가입할때부터 아이디가 공백이 아닌상태로 넘어왔다고 생각했는데 로그인뷰에서 다시 조건을 달아주신 이유가 궁금합니답!

sonms replied:
오 저는 좋은 의견이신 것 같아요 아무 생각없이 조건을 넣었네요 ㅋㅋㅋ ... 없어도 괜찮을 것 같네요!

sohee6989 commented on app/src/main/java/com/sopt/dive/core/designsystem/component/DiveSoptTextField.kt:
```diff
@@ -0,0 +1,94 @@
+    val colorBrush = remember {
+        Brush.linearGradient(
+            colors = listOf(
+                Color.Red,
```
우와 그라데이션 쓰셨네요~~ㅎㅎ
ui에 신경쓰시는 조장님이시군요~~~ㅋ.ㅋ

sohee6989 commented on app/src/main/java/com/sopt/dive/core/designsystem/component/DiveSoptTextField.kt:
```diff
@@ -0,0 +1,94 @@
+    }
+
+    TextField(
+        state = state,
```
이거 state는 값이랑 값 변경까지 한 번에 처리해주고 있는건가요?????
TextField 기본 속성이에요???

sonms replied:
이거는 혜민님 코멘트에도 달아 드렸는데 
[https://developer.android.com/develop/ui/compose/text/user-input?textfield=state-based&hl=ko#manage-tf-state](https://developer.android.com/develop/ui/compose/text/user-input?textfield=state-based&hl=ko#manage-tf-state) 요거 읽어보시면 좋을 것 가타요 ㅎㅎ 물론 파트 아티클에도 적어드릴게요!

sohee6989 commented on app/src/main/java/com/sopt/dive/ui/main/MainActivity.kt:
```diff
@@ -0,0 +1,188 @@
+
+        DiveSoptButton(
+            text = "로그아웃",
+            onClickButton = {
```
클릭에 대한 처리를 Activity에서 해주실 거면 여기서 람다로 감싸줄 필요가 없지 않나요??

sonms replied:
맞네요! 그냥 선언해줘도 되겠네요 역시 소희님 눈썰미 ..

sohee6989 commented on app/src/main/java/com/sopt/dive/ui/signin/SignInActivity.kt:
```diff
@@ -0,0 +1,277 @@
+}
+
+@Composable
+fun SignInScreen(
```
요기 접근제한자 붙여주시는건 어떠실까요??

sohee6989 commented on app/src/main/java/com/sopt/dive/ui/signup/SignUpActivity.kt:
```diff
@@ -0,0 +1,248 @@
+@Composable
+fun SignUpScreen(
+    paddingValues: PaddingValues,
+
```
이 공백 뭐죠...??????????????????😱
바로 밑에도.....

sonms replied:
하핫 가독성이라고 봐주시면 감사하겠습니다 ㅋㅋ... 파라미터 지우다가 빼먹은 거 같기도합니다..ㅠㅠ

sohee6989 commented on app/src/main/java/com/sopt/dive/data/remote/AuthManager.kt:
```diff
@@ -0,0 +1,75 @@
+        }
+    }
+
+    // Todo : 나중에 묶기
```
지금 해주세요.해주세요.ㅎㅎ

sohee6989 commented on app/src/main/java/com/sopt/dive/core/extension/ModifierExtensions.kt:
```diff
@@ -0,0 +1,97 @@
+    )
+}
+// 극한의 성능 최적화 시
+/*
```
이거 주석처리 코드 궁금합니다..!!!
설명해주시면 안된나요..??ㅎ.ㅎ
위에 noRipple이랑 뭐가 다른지 궁금합니다!!

sonms replied:
소희님 코드에도 적어놨는데 Modifier에서 noRipple을 사용할 시 @Composable방식과 composed 방식이 있는데 그 차이와 장단점, 문제를 확인하시면 좋을 것 같아요! 이 방식은 Modifier.node 방식에 대해 검색하시면 더 좋을 것 같아요 ㅎㅎ

angryPodo replied:
Modifier.node 어려워잉

sohee6989 reviewed (approved):
수고하셨습니다!!
코드 읽으면서 많이 배워갑니다~~🙇‍♀️
